protocols/relay/src/v2/relay/handler.rs

```diff
@@ -0,0 +1,892 @@
+                .finish(),
+            Event::OutboundConnectNegotiated {
+                circuit_id,
+                src_peer_id,
+                src_connection_id,
+                inbound_circuit_req: _,
+                dst_handler_notifier: _,
+                dst_stream: _,
+                dst_pending_data: _,
+            } => f
+                .debug_struct("Event::OutboundConnectNegotiated")
+                .field("circuit_id", circuit_id)
+                .field("src_peer_id", src_peer_id)
+                .field("src_connection_id", src_connection_id)
+                .finish(),
+            Event::OutboundConnectNegotiationFailed {
+                circuit_id,
+                src_peer_id,
+                src_connection_id,
+                inbound_circuit_req: _,
+                status,
+            } => f
+                .debug_struct("Event::OutboundConnectNegotiationFailed")
+                .field("circuit_id", circuit_id)
+                .field("src_peer_id", src_peer_id)
+                .field("src_connection_id", src_connection_id)
+                .field("status", status)
+                .finish(),
+            Event::CircuitClosed {
+                circuit_id,
+                dst_peer_id,
+                error,
+            } => f
+                .debug_struct("Event::CircuitClosed")
+                .field("circuit_id", circuit_id)
+                .field("dst_peer_id", dst_peer_id)
+                .field("error", error)
+                .finish(),
+        }
+    }
+}
+
+pub struct Prototype {
+    pub config: Config,
+}
+
+impl IntoProtocolsHandler for Prototype {
+    type Handler = Handler;
+
+    fn into_handler(self, _remote_peer_id: &PeerId, endpoint: &ConnectedPoint) -> Self::Handler {
+        Handler {
+            remote_addr: endpoint.get_remote_address().clone(),
+            config: self.config,
+            queued_events: Default::default(),
+            pending_error: Default::default(),
+            reservation_accept_futures: Default::default(),
+            reservation_deny_futures: Default::default(),
+            circuit_accept_futures: Default::default(),
+            circuit_deny_futures: Default::default(),
+            alive_lend_out_substreams: Default::default(),
+            circuits: Default::default(),
+            active_reservation: Default::default(),
+            keep_alive: KeepAlive::Yes,
+        }
+    }
+
+    fn inbound_protocol(&self) -> <Self::Handler as ProtocolsHandler>::InboundProtocol {
+        inbound_hop::Upgrade {
+            reservation_duration: self.config.reservation_duration,
+            max_circuit_duration: self.config.max_circuit_duration,
+            max_circuit_bytes: self.config.max_circuit_bytes,
+        }
+    }
+}
+
+/// [`ProtocolsHandler`] that manages substreams for a relay on a single
+/// connection with a peer.
+pub struct Handler {
+    remote_addr: Multiaddr,
+
+    /// Static [`Handler`] [`Config`].
```

Comment:
```suggestion
    // Static [`Handler`] [`Config`].
```
Not a public type. Also for the other code docs here.

Reply:
Do I understand correctly that you are suggesting to write `//` instead of `///` as this is not a public type and thus won't be picked up by `cargo doc` in the first place?

Thus far I simply followed the convention of using `///` on all struct properties, no matter whether they are public or not. If I am not missing something, this is the convention used across the entire code base.